@costing This gives us messages of the following form:
[ERROR] Device EfficiencyGlobal (30250) had at least one message above severity ERROR: Invalid URL : http://alice-ccdb.cern.ch:80/Users/jgrosseo/efficiency/simple
Apart from the PID in brackets, we don't need to remove anything and can aggregate.
What do you think?

--- a/Framework/Core/src/runDataProcessing.cxx
+++ b/Framework/Core/src/runDataProcessing.cxx
@@ -274,7 +274,7 @@ bool areAllChildrenGone(std::vector<DeviceInfo>& infos)
 /// Calculate exit code
 int calculateExitCode(DeviceSpecs& deviceSpecs, DeviceInfos& infos)
 {
-  std::regex regexp("^\\[([\\d+:]*)\\]\\[\\w+\\] ");
+  std::regex regexp(R"(^\[([\d+:]*)\]\[\w+\] )");
   int exitCode = 0;
   for (size_t di = 0; di < deviceSpecs.size(); ++di) {
     auto& info = infos[di];

